customized_list_view/models/custom_list.py

```diff
@@ -0,0 +1,139 @@
+    original_arch = fields.Text(readonly=True)
+
+    @api.constrains("model_id")
+    def _check_unique_model(self):
+        for rec in self:
+            same_model_recs = self.env["custom.list.view"].search(
+                [
+                    ("model_id", "=", rec.model_id.id),
+                    ("id", "!=", rec.id),
+                ]
+            )
+            if same_model_recs:
+                raise exceptions.ValidationError(
+                    _(
+                        "Only one record per model is allowed. "
+                        "Please modify existing record instead."
+                    )
+                )
+
+    @api.depends("model_id")
+    def _compute_model_name(self):
+        for rec in self:
+            rec.model_name = rec.model_id.model
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super().create(vals_list)
+        res.original_arch = res.list_view_id.arch
+        return res
+
+    def write(self, vals):
+        res = super().write(vals)
+        return res
+
+    def button_apply_changes(self):
+        self.ensure_one()
+        doc = etree.XML(self.original_arch)
+        for mod_line in self.line_ids:
+            target = mod_line.before.name if mod_line.before else mod_line.after.name
+            for node in doc.xpath("/tree/field[@name='%s']" % target):
+                node_string = (
+                    "<field name='%s' widget='%s' optional='%s' string='%s'/>"
+                    % (
+                        mod_line.field_id.name,
+                        mod_line.widget or "",
+                        mod_line.optional or "",
+                        mod_line.label or mod_line.field_id.field_description,
+                    )
+                )
+                new_node = etree.fromstring(node_string)
+                if mod_line.before:
+                    node.addprevious(new_node)
+                else:
+                    node.addnext(new_node)
+        new_arch = etree.tostring(doc, encoding="unicode").replace("\t", "")
+        self.list_view_id.arch = new_arch
+
+    def button_roll_back(self):
+        self.ensure_one()
+        doc = etree.XML(self.original_arch)
+        new_arch = etree.tostring(doc, encoding="unicode").replace("\t", "")
+        self.list_view_id.arch = new_arch
+
+
+class CustomListViewLine(models.Model):
+    _name = "custom.list.view.line"
+    _description = "Custom List View Line"
+
+    custom_list_view_id = fields.Many2one("custom.list.view")
+    model_id = fields.Many2one(related="custom_list_view_id.model_id")
+    model_name = fields.Char(related="custom_list_view_id.model_name")
+    field_id = fields.Many2one("ir.model.fields", "New Field")
+    after = fields.Many2one("ir.model.fields", "Place After")
+    before = fields.Many2one("ir.model.fields", "Place Before")
+    optional = fields.Selection(
+        [("show", "show"), ("hide", "hide"), ("", "")], default=""
```

Comment:
Why do you added empty option?
If you remove `default` argument and empty option then by default in selection empty option is auto added.